### ISSUE TITLE
Initial model structure (when resolution = 4x4)

### ISSUE THREAD

**Issue by nashory** [nashory/progressive-growing-torch#14]
For test:
~~~
Generator structure:
nn.Sequential {
  [input -> (1) -> (2) -> output]
  (1): nn.Sequential {
    [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> output]
    (1): nn.SpatialFullConvolution(512 -> 512, 4x4)
    (2): nn.LeakyReLU(0.2)
    (3): nn.SpatialCrossMapLRN
    (4): nn.SpatialFullConvolution(512 -> 512, 3x3, 1,1, 1,1)
    (5): nn.LeakyReLU(0.2)
    (6): nn.SpatialBatchNormalization (4D) (512)
  }
  (2): nn.Sequential {
    [input -> (1) -> (2) -> (3) -> output]
    (1): nn.SpatialFullConvolution(512 -> 3, 1x1)
    (2): nn.LeakyReLU(0.2)
    (3): nn.SpatialCrossMapLRN
  }
}
Discriminator structure:
nn.Sequential {
  [input -> (1) -> (2) -> output]
  (1): nn.Sequential {
    [input -> (1) -> (2) -> (3) -> output]
    (1): nn.SpatialConvolution(3 -> 512, 1x1)
    (2): nn.LeakyReLU(0.2)
    (3): nn.SpatialBatchNormalization (4D) (512)
  }
  (2): nn.Sequential {
    [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> output]
    (1): nn.MinibatchStatConcat
    (2): nn.SpatialConvolution(513 -> 512, 3x3, 1,1, 1,1)
    (3): nn.LeakyReLU(0.2)
    (4): nn.SpatialCrossMapLRN
    (5): nn.SpatialConvolution(512 -> 512, 4x4)
    (6): nn.LeakyReLU(0.2)
    (7): nn.SpatialCrossMapLRN
    (8): nn.View(512)
    (9): nn.Linear(512 -> 1)
    (10): nn.Sigmoid
  }
}
~~~

**Comment by zhangqianhui** [comment]
hello, why the original paper remove tanh function ? Do you understand?

**Comment by nashory** [comment]
Hi, I don't know why because no mention about that in the paper.
As far as I know, clipping using TanH() helps stabilizing the training, but the authors did not use it.
Perhaps they used various kinds of normalization techniques, such as WeightNorm or PixelNorm and these custom layers might have helped the stable training even without clipping... it's just my guessing.
